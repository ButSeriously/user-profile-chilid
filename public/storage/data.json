{
  "likes": ["Likes", 121],
  "following": ["Following", 723],
  "followers": ["Followers", 4433],
  "user_photo": "./img/user-photo.jpg",
  "user_name": "Harvey Specter",
  "user_location": "New York, USA",
  "comments": [
    {
      "photo": "./img/user-photo.jpg",
      "name": "Mike Ross",
      "content": "Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.",
      "date": "2017-10-22T07:11:00.611Z"
    },
    {
      "photo": "./img/user-photo.jpg",
      "name": "Mike Ross",
      "content": "Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.",
      "date": "2017-10-20T12:45:03.564Z"
    }
  ],
  "logged_in": {
    "photo": "./img/user-photo.jpg",
    "user": "Mike Ross"
  },
  "link": "https://butseriously.github.io/user-profile-chilid/"
}