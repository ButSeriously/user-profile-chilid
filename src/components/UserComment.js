import PropTypes from 'prop-types';
import React from 'react';
import moment from 'moment';

class UserComment extends React.Component {
  timeDifference = (prevDate) => {
    const current = moment();
    const previous = moment(prevDate);
    return current.from(previous, true);
  };
  render() {
    return (
      <div className="user_comments__container">
        <div className="user_comments__photo" style={{ backgroundImage: `url(${this.props.comment.photo})` }} alt="user.jpg" />
        <div className="user_comments__content">
          <p className="user_comments__name">
            { this.props.comment.name }
          </p>
          <p className="user_comments__commentary">
            { this.props.comment.content }
          </p>
        </div>
        <p className="user_comments__date">
          { this.timeDifference(this.props.comment.date) }
        </p>
      </div>
    );
  }
}

UserComment.propTypes = {
  comment: PropTypes.object.isRequired,
};

export default UserComment;
