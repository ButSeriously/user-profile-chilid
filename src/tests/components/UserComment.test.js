import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import UserComment from '../../components/UserComment';
import data from '../fixtures/comments.json';

test('should render UserComment', () => {
  const comment = data.comments[0];
  comment.date = moment().add(3, 'days');
  const wrapper = shallow(<UserComment comment={comment} />);
  expect(wrapper).toMatchSnapshot();
});
