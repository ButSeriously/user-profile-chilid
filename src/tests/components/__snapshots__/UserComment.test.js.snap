// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserComment 1`] = `
<div
  className="user_comments__container"
>
  <div
    alt="user.jpg"
    className="user_comments__photo"
    style={
      Object {
        "backgroundImage": "url(./img/user-photo.jpg)",
      }
    }
  />
  <div
    className="user_comments__content"
  >
    <p
      className="user_comments__name"
    >
      Mike Ross
    </p>
    <p
      className="user_comments__commentary"
    >
      Lorem ipsum dolor sit amet enim. Etiam ullamcorper. Suspendisse a pellentesque dui, non felis. Maecenas malesuada elit lectus felis, malesuada ultricies. Curabitur et ligula.
    </p>
  </div>
  <p
    className="user_comments__date"
  >
    3 days
  </p>
</div>
`;
